server/memphis_handler_schemas.go

```diff
@@ -0,0 +1,368 @@
+	}
+	opts := options.Update().SetUpsert(true)
+	updateResults, err := schemasCollection.UpdateOne(context.TODO(), filter, update, opts)
+	if err != nil {
+		serv.Errorf("CreateSchema error: " + err.Error())
+		c.AbortWithStatusJSON(500, gin.H{"message": "Server error"})
+		return
+	}
+	if updateResults.MatchedCount > 0 {
+		serv.Warnf("Scheam with the same name is already exist")
+		newSchema.ID = schema.ID
+	} else {
+		message := "Schema " + schemaName + " has been created"
+		serv.Noticef(message)
+		var auditLogs []interface{}
+		newAuditLog := models.AuditLog{
+			ID:            primitive.NewObjectID(),
+			StationName:   schemaName,
+			Message:       message,
+			CreatedByUser: user.Username,
+			CreationDate:  time.Now(),
+			UserType:      user.UserType,
+		}
+		auditLogs = append(auditLogs, newAuditLog)
+		err = CreateAuditLogs(auditLogs)
+		if err != nil {
+			serv.Warnf("CreateSchema error: " + err.Error())
+		}
+	}
+
+	filter = bson.M{"_id": newSchemaVersion.ID, "is_deleted": false}
+	update = bson.M{
+		"$setOnInsert": bson.M{
+			"_id":             newSchemaVersion.ID,
+			"version_number":  newSchemaVersion.VersionNumber,
+			"active":          newSchemaVersion.Active,
+			"created_by_user": newSchemaVersion.CreatedByUser,
+			"creation_date":   newSchemaVersion.CreationDate,
+			"schema_content":  newSchemaVersion.SchemaContent,
+		},
+	}
+	_, err = schemasVersionCollection.UpdateOne(context.TODO(), filter, update, opts)
+
+	if err != nil {
+		serv.Errorf("CreateSchema error: " + err.Error())
+		c.AbortWithStatusJSON(500, gin.H{"message": "Server error"})
+		return
+	}
+	newSchema.Versions = append(newSchema.Versions, newSchemaVersion.ID)
+	c.IndentedJSON(200, newSchema)
+}
+
+func (sh SchemasHandler) GetAllSchemas(c *gin.Context) {
+	schemas, err := sh.GetAllSchemasDetails()
+	if err != nil {
+		serv.Errorf("GetAllSchemas error: " + err.Error())
+		c.AbortWithStatusJSON(500, gin.H{"message": "Server error"})
+		return
+	}
+	c.IndentedJSON(200, schemas)
+}
+
+func (sh SchemasHandler) GetSchemaDetails(c *gin.Context) {
+	var body models.GetSchemaDetails
+	ok := utils.Validate(c, &body, false, nil)
+	if !ok {
+		return
+	}
+	schemaName := strings.ToLower(body.SchemaName)
+	schemaDetails, err := sh.GetSchemaDetailsBySchemaName(schemaName, body.VersionNumber)
+
+	if err != nil {
+		serv.Errorf("GetSchemaDetails error: " + err.Error())
+		c.AbortWithStatusJSON(500, gin.H{"message": "Server error"})
+		return
+	}
+	c.IndentedJSON(200, schemaDetails)
+}
+
+func (sh SchemasHandler) RemoveSchema(c *gin.Context) {
+	var body models.RemoveSchema
+	ok := utils.Validate(c, &body, false, nil)
+	if !ok {
+		return
+	}
+	schemaName := strings.ToLower(body.SchemaName)
+	err := sh.FindAndDeleteSchema(schemaName)
+
+	if err != nil {
+		serv.Errorf("RemoveSchema error: " + err.Error())
+		c.AbortWithStatusJSON(500, gin.H{"message": "Server error"})
+		return
+
+	}
+	c.IndentedJSON(200, []string{})
```

Comment:
why are you returning an array? please return an empty object, take reference from tther delete operations